suncalc.js

```diff
@@ -43,6 +43,14 @@ function altitude(H, phi, dec) { return asin(sin(phi) * sin(dec) + cos(phi) * co
 
 function siderealTime(d, lw) { return rad * (280.16 + 360.9856235 * d) - lw; }
 
+function astroRefraction(h) {
+    if (h < 0) // the following formula works for positive altitudes only.
+        h = 0; // if h = -0.08901179 a div/0 would occur.
+
+    // formula 16.4 of "Astronomical Algorithms" 2nd edition by Jean Meeus (Willmann-Bell, Richmond) 1998.
+    // 1.02 / tan(h + 10.26 / (h + 5.10)) h in degrees, result in arc minutes -> converted to rad:
+    return 0.0002967 / Math.tan(h + 0.00312536 / (h + 0.08901179));
```

Comment:
The formula  1.02 / tandec(h + 10.26 / (h + 5.10)) is in degrees and the result is in arc minutes
(The tangens in this formula calcs degrees and not in radians!)

Conversion to rad:

hrad = h * rad

(1.02*rad/60) / tanrad(rad * (h + 10.26 / (h + 5.10))                      // div 60 is for conversion from minutes to degree resp. rad
(1.02*rad/60) / tanrad(rad * h + rad*(10.26 / (h + 5.10)))  
(1.02*rad/60) / tanrad(rad * hrad/rad + rad*(10.26 / (hrad/rad + 5.10)))   // replace h with hrad/rad

(1.02*rad/60) / tanrad(hrad + rad*(rad*10.26 / rad*(hrad/rad + 5.10)))     // simplify 
(1.02*rad/60) / tanrad(hrad + rad*rad*10.26 / hrad + 5.10*rad)))           // simplify
 
 0.000296706 / tanrad(hrad + 0.003125375 / hrad + 0.089011792)))           // calc constants




> Am 08.01.2016 um 15:47 schrieb Vladimir Agafonkin <notifications@github.com>:
> 
> In suncalc.js <https://github.com/mourner/suncalc/pull/64#discussion_r49194462>:
> 
> > @@ -43,6 +43,14 @@ function altitude(H, phi, dec) { return asin(sin(phi) * sin(dec) + cos(phi) * co
> >  
> >  function siderealTime(d, lw) { return rad * (280.16 + 360.9856235 * d) - lw; }
> >  
> > +function astroRefraction(h) {
> > +    if (h < 0) // the following formula works for positive altitudes only.
> > +        h = 0; // if h = -0.08901179 a div/0 would occur.
> > +
> > +    // formula 16.4 of "Astronomical Algorithms" 2nd edition by Jean Meeus (Willmann-Bell, Richmond) 1998.
> > +    // 1.02 / tan(h + 10.26 / (h + 5.10)) h in degrees, result in arc minutes -> converted to rad:
> > +    return 0.0002967 / Math.tan(h + 0.00312536 / (h + 0.08901179));
> Can you briefly explain why the previous formula was wrong? Are the new constant values the same as rad * 10.26 etc.?
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/mourner/suncalc/pull/64/files#r49194462>.
> 

> 	 1.02 / tan(h + 10.26 / (h + 5.10))
>